openassessment/workflow/models.py

```diff
@@ -316,17 +316,20 @@ def update_from_assessments(self, assessment_requirements):
 
         new_staff_score = self.get_score(assessment_requirements, {'staff': step_for_name.get('staff', None)})
         if new_staff_score:
-            old_score = self.score
-            if not old_score or old_score['points_earned'] != new_staff_score['points_earned']:
+            old_score = sub_api.get_latest_score_for_submission(self.submission_uuid)
+            if not old_score or old_score['created_at'] != new_staff_score['created_at']:
```

Comment:
Scores don't have a UID, but `created_at` should be a good enough approximation to determine whether we need to update.